### PR TITLE
Flubu build script

### PR DESCRIPTION
**Pull request by mzorec**
Versioning: All projects currently gets version from EasyCaching.ProjectVersion.txt. If you want this can be changed back to version.props but it's a bit harder to make publishing of nuget packages through build script. It seems to me that this is the type of solution that all projects should have same version.

Nuget publishing: Nuget package will be published automatically when you increase version in EasyCaching.ProjectVersion.txt as I said this can be changed to your preferences. Before nuget publishing will work you have to add nuget api key as environment variable in travis (don't worry it won't be publicly visible.)
name of the environemt variable must be 'apiKey' you can add it in travis -> More Options -> Settings -> Environment variables

If u have any questions fell free to ask or if I wan't any changes in build script just tell.

P.S. In next days i'll create another pull request where I will put docker commands in build script.

### CODE REVIEW

**Comment by catcherwong** [comment]
> Versioning: All projects currently gets version from EasyCaching.ProjectVersion.txt. If you want this can be changed back to version.props but it's a bit harder to make publishing of nuget packages through build script. It seems to me that this is the type of solution that all projects should have same version.

Sometimes, a little change for one package, I am not sure that if we update all packages' version is friendly to the users or not.

**Comment by catcherwong** [comment]
@mzorec All things LGTM. 

Could you please change the base branch from `master` to `dev`?

File: build/BuildScript.cs [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+      
+            var build = context.CreateTarget("Build")
+                .SetDescription("Build's the solution.")
+                .AddCoreTask(x => x.Restore())
+                .AddCoreTask(x => x.UpdateNetCoreVersionTask(_easyCachingProjects))
+                .AddCoreTask(x => x.Build())
+                .DependsOn(buildVersion);
+
+            var runTests = context.CreateTarget("Run.Tests")
+                .SetDescription("Run's all Easy caching tests.")
```
Comment (catcherwong) [inline build/BuildScript.cs]: maybe the description should be `Run's all EasyCaching tests.`

**Comment by mzorec** [comment]
> Could you please change the base branch from `master` to `dev`?

Ok

> > Versioning: All projects currently gets version from EasyCaching.ProjectVersion.txt. If you want this can be changed back to version.props but it's a bit harder to make publishing of nuget packages through build script. It seems to me that this is the type of solution that all projects should have same version.
> 
> Sometimes, a little change for one package, I am not sure that if we update all packages' version is friendly to the users or not.

In my opinion it is better if you update just one package. Usually package owners don't bother with this tho. I'll change build script to use version.props file.

**Pull request closed without merging**